packages/flutter/lib/src/material/slider.dart

```diff
@@ -34,6 +34,45 @@ typedef PaintValueIndicator = void Function(PaintingContext context, Offset offs
 
 enum _SliderType { material, adaptive }
 
+/// Possible ways a user can interact with a [Slider].
+enum SliderInteraction {
+  /// Allows the user to interact with a [Slider] by tapping or sliding anywhere
+  /// on the track.
+  ///
+  /// Essentially all possible interactions are allowed.
+  ///
+  /// This is different from [SliderInteraction.slideOnTrack] as when you try
+  /// to slide anywhere other than the thumb, the thumb will move to the first
+  /// point of contact.
+  tapAndSlide,
+  /// Allows the user to interact with a [Slider] by only tapping anywhere on
+  /// the track.
+  ///
+  /// Sliding interaction is ignored.
+  tapOnly,
+  /// Allows the user to interact with a [Slider] only by sliding anywhere on
+  /// the track.
+  ///
+  /// Tapping interaction is ignored.
+  slideOnTrack,
+  /// Allows the user to interact with a [Slider] only by sliding the thumb.
+  ///
+  /// Taping and sliding interactions on the track are ignored.
+  slideThumb,
```

Comment:
What if both of these is desired?

Reply:
`slideOnTrack` is already a superset of `slideThumb`, as thumb is the part of the track. will add this to the docs anyway.
`slideOnTrack` U `slideThumb` = `slideOnTrack`.
I can rename `slideOnTrack` with `slideOnly` if that makes it clearer.

Reply:
Just tested all the interactions. I think `slideOnly` makes perfect sense. 

I've some concerns about side effects such, as these behavior issues might already exist in the slider widget but when we restrict another type of interaction, these existing behavior issues become more obvious than before. 

For instance,  when using`slideOnly` I cannot tap and only slide this makes it obvious Slider can be dragged from outside the window or continue dragging even if the cursor leaves the window. This is matching scrollbar drag behavior which was also draggable from outside the window, until a few months ago.

With `slideThumb`, I can tap within the overlay area to adjust the value without sliding the thumb, Where I would expect to only slide (or drag action to work), 

Here I created a code sample 


<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: const SliderExample(),
    );
  }
}

class SliderExample extends StatefulWidget {
  const SliderExample({super.key});

  @override
  State<StatefulWidget> createState() => _SliderExampleState();
}

class _SliderExampleState extends State<SliderExample> {
  double slider1 = 0.0;
  double slider2 = 0.0;
  double slider3 = 0.0;
  double slider4 = 0.0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Slider Sample'),
      ),
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            children: [
              const Text('tapAndSlide - default'),
              Slider(
                value: slider1,
                onChanged: (double value) {
                  setState(() {
                    slider1 = value;
                  });
                },
              ),
              const Text('tapOnly'),
              Slider(
                allowedInteraction: SliderInteraction.tapOnly,
                value: slider2,
                onChanged: (double value) {
                  setState(() {
                    slider2 = value;
                  });
                },

              ),
              const Text('slideOnly (slideOnTrack)'),
              Slider(
                allowedInteraction: SliderInteraction.slideOnTrack,
                value: slider3,
                onChanged: (double value) {
                  setState(() {
                    slider3 = value;
                  });
                },

              ),
              const Text('slideThumb'),
              Slider(
                allowedInteraction: SliderInteraction.slideThumb,
                value: slider4,
                onChanged: (double value) {
                  setState(() {
                    slider4 = value;
                  });
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}

``` 
	
</details>


https://user-images.githubusercontent.com/48603081/222394293-8b665e7a-e352-4d96-899a-e8368aa8ddca.mov

Reply:
thanks for pointing this out. this is an issue.

i realise `slideOnly` should be like [the video](https://drive.google.com/file/d/1kYEIxFB02E-avJCYsk5KWUPtANqzP_C_/view?usp=share_link) the issue author added. it can be argued that in its current behavior, `slideOnly` goes against `Slider`'s basic spec that the thumb for changing value and not the track.

https://user-images.githubusercontent.com/44755140/222881471-2fdf70f9-f23e-464b-b035-79746253c109.mov

also, thanks for finding the `slideThumb` bug, will fix it.

Reply:
fixed the `slideThumb` bug.

for the `slideOnly` issue, the above video works close to as intended on macos but not on android. this is because a tap down on an android screen is registered as a `dragStart`, a few `dragUpdate`s and a `dragEnd`, which is weird. you can test this by adding logs on all `_handle*` functions in the `slider.dart` file.
a plausible solution to this is to have a threshold for when the *slide* should start, but it needs to be eye balled as there is no material spec for this. thoughts?

Reply:
@TahaTesser could you help me with this?

Reply:
This looks good!